File: research/stellar/examples/javascript/account/transfer.js
const StellarSdk = require('stellar-sdk')
const server = new StellarSdk.Server('https://horizon-testnet.stellar.org')

var argv = require('yargs')
    .usage('Usage: $0 --sourceKey [string] --destinationAddress [string] --amount [string] --asset format: `code:issuer`')
    .demandOption(['sourceKey', 'destinationAddress', 'amount'])
    .option('asset', {
      alias: 'a',
      type: 'string',
      description: 'format: code:issuer',
      default: ''
    })
    .argv;


async function transfer(sourceKey, destination, amount, asset) {
  const sourceKeypair = StellarSdk.Keypair.fromSecret(sourceKey)
  const account = await server.loadAccount(sourceKeypair.publicKey())

  const fee = await server.fetchBaseFee()

  if (asset === '') {
    asset = StellarSdk.Asset.native()
  } else {
    assetStr = asset.split(':')
    if (assetStr.length !== 2) throw new Error('asset has wrong format')
    asset = new StellarSdk.Asset(assetStr[0], assetStr[1])
  }

  const transaction = new StellarSdk.TransactionBuilder(account, {
      fee,
      networkPassphrase: StellarSdk.Networks.TESTNET
    })
    // Add a payment operation to the transaction
    .addOperation(StellarSdk.Operation.payment({
      destination,
      // The term native asset refers to lumens
      asset,
      amount: amount.toString(),
    }))
    .setTimeout(30)
    .build()

  transaction.sign(sourceKeypair)

  console.log(transaction.toEnvelope().toXDR('base64'))

  try {
    const transactionResult = await server.submitTransaction(transaction)
    console.log(JSON.stringify(transactionResult, null, 2))
    console.log('\nSuccess! View the transaction at: ')
    console.log(transactionResult._links.transaction.href)
  } catch (e) {
    console.log('An error has occured:')
    console.log(e)
  }
}

transfer(argv.sourceKey, argv.destinationAddress, argv.amount, argv.asset)

File: research/stellar/examples/python/account/transfer.py
# pylint: disable=no-value-for-parameter
from stellar_sdk import Server, Keypair, TransactionBuilder, Network
from stellar_sdk.exceptions import BadRequestError
import click

@click.command()
@click.option('--sourcekey', help='Secret key of the source', required=True)
@click.option('--destinationaddress', help='Destination address of the account to activate', required=True)
@click.option('--amount', help='Amount to transfer', required=True, type=int)
@click.option('--asset', help='Asset format code:issuer', default='')

def transfer(sourcekey, destinationaddress, amount, asset):
  server = Server(horizon_url="https://horizon-testnet.stellar.org")
  source_keypair = Keypair.from_secret(sourcekey)
  source_public_key = source_keypair.public_key
  source_account = server.load_account(source_public_key)

  base_fee = server.fetch_base_fee()

  issuer = None

  if asset == '':
    asset = 'XLM'
  else:
    assetStr = asset.split(':')
    if len(assetStr) != 2:
      return Exception('Wrong asset format')
    asset = assetStr[0]
    issuer = assetStr[1]

  transaction = (
      TransactionBuilder(
          source_account=source_account,
          network_passphrase=Network.TESTNET_NETWORK_PASSPHRASE,
          base_fee=base_fee,
      )
          .append_payment_op(destination=destinationaddress, amount=str(amount), asset_code=asset, asset_issuer=issuer)
          .set_timeout(30)
          .build()
  )

  transaction.sign(source_keypair)

  print(transaction.to_xdr())

  try:
    response = server.submit_transaction(transaction)
    print("Transaction hash: {}".format(response["hash"]))
    print(response)
  except BadRequestError as e:
    print(e)

if __name__ == '__main__':
  transfer()